packages/next/src/client/components/app-router.tsx

```diff
@@ -645,9 +645,13 @@ function Router({
         require('./dev-root-not-found-boundary').DevRootNotFoundBoundary
       content = (
         <DevRootNotFoundBoundary>
-          <MissingSlotContext.Provider value={missingSlots}>
-            {content}
-          </MissingSlotContext.Provider>
+          {missingSlots ? (
+            <MissingSlotContext.Provider value={missingSlots}>
+              {content}
+            </MissingSlotContext.Provider>
```

Comment:
don't necessarily need to change this now but since this value is set once for the lifetime of the app we don't really need to use context to make it available. It can just be a global value. The risk of this setup is if it somehow ever did update and no longer exist we'd unmount the tree here when we remove the context provider.

Also pretty sure you no longer need `.Provider` and can just render the context since that was added to React 19.

Reply:
Turns out this was added because each nested `LayoutRouter` renders its own `NotFoundBoundary`, so we need to make sure all of them have the `missingSlots` data available to properly warn. I could expose `missingSlots` to the layout router but it feels a bit messier than doing it this way.

I've removed the `.Provider` though 👍